md/pages-development.md

```diff
@@ -0,0 +1,143 @@
+
+### PageController interface
+```java
+public interface PageController {
+
+/**
+* Let the custom page parse request for specific attribute handling.
+*
+* @param request the HTTP servlet request intended to be used as in a servlet
+* @param response the HTTP servlet response intended to be used as in a servlet
+* @param pageResourceProvider provide access to the resources contained in the custom page zip
+* @param pageContext provide access to the data relative to the context in which the custom page is displayed
+*/
+public void doGet(HttpServletRequest request, HttpServletResponse response, PageResourceProvider pageResourceProvider, PageContext pageContext);
+}
+```
+
+### Permissions for custom pages
+
+If you are using [REST API authorization](rest-api-authorization.md) and your custom page is an HTML page using the Web REST API, 
+you need to specify the permissions that a user needs to have in order to access the resources in the custom page. 
+These permissions are defined in the `page.properties` file. If your custom page is written in Groovy and uses the Bonita Engine Java APIs, you do not need to specify permissions.
+
+For each REST resource accessed in the page, specify the authorization needed for each method used.  
+You can find examples of the default resources in [`resources-permissions-mapping.properties`](BonitaBPM_platform_setup.md).
+
+The following example shows the permissions defined for a custom page that enables a user to view but not update organization information:
+
+```
+#The name must start with 'custompage_'
+name=custompage_orgViewer
+displayName=Organization viewer
+description=Organization viewer page. You cannot modify the organization from this page.
+resources=[GET|identity/user, GET|identity/personalcontactdata, GET|identity/professionalcontactdata, GET|identity/role, GET|identity/group, GET|identity/membership, GET|customuserinfo/user, GET|customuserinfo/definition, GET|customuserinfo/value] 
+```
+
+### Custom page examples
+
+Two custom page examples are available in Bonita Portal. Both examples show how to:
+
+* Get the Bonita CSS
+* Write simple HTML code
+* Get session information, including the locale
+* Get resources from the custom page definition
+* Call the Engine Java APIs (in the Groovy example) or the Web REST APIs (in the.md example)
+* Write a clickable link to an external page
+* Write a clickable link to a portal page
+* Write locale-specific messages
+
+**Groovy example page** defines a custom page using Groovy. 
+**HTML example page** defines a custom page using only HTML. 
+In practice, you will probably use a combination of Groovy and HTML to create your custom pages.
+
+To view an example page, map the page to an [application](applications.md) so that you can view it in Bonita Portal. To view the source of an example, [export the custom page](resource-management.md#export).
+
+### Reuse Bonita Portal content
```

Comment:
Do we want to keep this section as we will change the process visu technology soon and this will no longer work ?

Reply:
yes. We are not sure of when we can imbed the bpmn-visu library